ext/app/electron/main.ts

```diff
@@ -1,511 +1,35 @@
-      // Add to list of recent docs in the menu
-      recentItems.addItem(filePath);
-      serverMethods.updateUserConfig({ recentItems: recentItems.listItems() });
-      // TODO: Electron does not yet support updating the menu except by reassigning the entire
-      // menu.  There are proposals to allow menu templates include callbacks that
-      // are called on menu open.  https://github.com/electron/electron/issues/528
-      appMenu.rebuildMenu();
-      electron.Menu.setApplicationMenu(appMenu.getMenu());
-    });
-
-    // serverMethods.onBackupMade((bakPath: string) => notifyMigrateBackup(bakPath));
-
-    // Check for updates, and check again periodically (if user declines, it's the interval till
-    // the next reminder, so too short would be annoying).
-    //    if (updateManager.startAutoCheck()) {
-    //      updateManager.schedulePeriodicChecks(6*3600);
-    //    } else {
-    //      log.warn("updateManager not starting (known not to work on Linux)");
-    //    }
-
-    // Now that we are ready, future 'open-file' events should just open windows directly.
-    this.app.removeAllListeners('open-file');
-    this.app.on('open-file', (_, filepath) => this.handleOpen(serverMethods, filepath));
-
-    this.app.on('will-quit', function(event) {
-      event.preventDefault();
-      shutdown.exit(0);
-    });
-
-    // Quit when all windows are closed.
-    this.app.on('window-all-closed', () => {
-      this.app.quit();
-    });
-
-    // Plugins create <webview> elements with a "plugins" partition; here we add a special header
-    // to all such requests. Requests for plugin content without this header will be rejected by
-    // the server, to ensure that untrusted content is only loaded in protected <webview> elements.
-    electron.session.fromPartition("plugins").webRequest.onBeforeSendHeaders((details, callback) => {
-      details.requestHeaders["X-From-Plugin-WebView"] = "true";
-      callback({requestHeaders: details.requestHeaders});
-    });
-  }
-
-  private reportError(e: Error) {
-    electron.dialog.showMessageBoxSync({
-      type: "info",
-      buttons: ["Ok"],
-      message: "Error",
-      detail: String(e)
-    });
-  }
-
-  private async normalizePath(filepath: string) {
-    // Use realpath if possible.
-    try {
-      filepath = await fse.realpath(filepath);
-    } catch (e) {
-      // if there's a problem, e.g. file doesn't exist or is symlink to
-      // nowhere, don't panic.
-    }
-    return path.normalize(filepath);
-  }
-}
-
-const gristApp = new GristApp();
-gristApp.main();
-
-
-// Set a default for an environment variable.
-function setDefaultEnv(name: string, value: string) {
-  const current = process.env[name]
-  if (current !== undefined) {
-    return current;
-  }
-  process.env[name] = value;
-  return value;
-}
-
-function reportErrorAndStop(e: Error) {
-  console.error(e);
-  process.exit(1);
-}
+loadConfig().then(() => {
+  new GristApp().main();
+});
```

Comment:
Can you add a `.catch` so it is clear locally that errors in loadConfig are handled.